src/main/java/org/springframework/data/couchbase/repository/query/AbstractN1qlBasedQuery.java

```diff
@@ -122,10 +122,6 @@ protected static N1qlQuery buildQuery(Statement statement, JsonValue queryPlaceh
   protected Object executeDependingOnType(N1qlQuery query, N1qlQuery countQuery, QueryMethod queryMethod,
       Pageable pageable, Class<?> typeToRead) {
 
-    if (queryMethod.isModifyingQuery()) {
```

Comment:
I don't think the check is even valid, commons always returns false https://github.com/spring-projects/spring-data-commons/blob/b5af2e9d9685f176d0a5ef0154baeb93b698a556/src/main/java/org/springframework/data/repository/query/QueryMethod.java#L215